### DISCUSSION TITLE
How to dig libJvm.so?

### DISCUSSION THREAD

**Discussion by BigBabolBee** [async-profiler/async-profiler#631] [Q&A] [answered]
Hi guys,

This tool is leveraging **AsyncGetCallTrace**  to perform stack unwinding, which is so much awesome. 

Since the function comes from a shared lib - libJvm.so, does anyone know how could we dig this powerful SO? I'm wondering how this function works and if there are other interesting functions to use.

Any link, docs or source code would be highly appreciated!

Thanks

**Comment by apangin** [comment, marked as answer]
`libjvm.so` is basically a compiled binary of the Java Virtual Machine.
There are multiple implementations of a JVM, most common are HotSpot and J9, with their source code available in [OpenJDK](https://github.com/openjdk/jdk) and [OpenJ9](https://github.com/eclipse-openj9/openj9/) repositories correspondingly.

Like with any other shared library, you can check what functions it exports:
```
nm -Dg --defined-only $JAVA_HOME/lib/server/libjvm.so
```
Keep in mind though, none of these symbols are a part of a public API, they may appear, disappear, and change semantics in any release.

**Reply by BigBabolBee** [reply]
It's so KIND of you!

This elaborate reply helps me a lot.

Thanks again.
